fabric-biomes-v1/src/main/java/net/fabricmc/fabric/mixin/biomes/MixinSetBaseBiomesLayer.java

```diff
@@ -0,0 +1,112 @@
+package net.fabricmc.fabric.mixin.biomes;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+
+import net.minecraft.class_604;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.layer.Layer;
+import net.minecraft.world.biome.layer.SetBaseBiomesLayer;
+import net.minecraft.world.gen.CustomizedWorldProperties;
+
+import net.fabricmc.fabric.api.biomes.v1.Climate;
+import net.fabricmc.fabric.impl.biomes.InternalBiomeUtils;
+import net.fabricmc.fabric.impl.biomes.LayerRandom;
+
+@Mixin(SetBaseBiomesLayer.class)
+public abstract class MixinSetBaseBiomesLayer extends Layer {
+	public MixinSetBaseBiomesLayer(long l) {
+		super(l);
+	}
+
+	@Shadow
+	private Biome[] field_2262;
+	@Shadow
+	private Biome[] field_2263;
+	@Shadow
+	private Biome[] field_2264;
+	@Shadow
+	private Biome[] field_2265;
+
+	@Shadow
+	@Final
+	private CustomizedWorldProperties field_2266;
+
+	/**
+	 * @author Valoeghese
+	 * @author too lazy to inject? no idea
```

Comment:
It would be absolutely cursed if done as injects, redirects, and modify variables.

Reply:
its fabric api

Reply:
api shud not break stuff